### ISSUE TITLE
EventHub - Documentataion / Functionality is not accurate

### ISSUE THREAD

**Issue by ssrahul96** [hashicorp/terraform-provider-azurerm#16606]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.9

### AzureRM Provider Version

3.3.0

### Affected Resource(s)/Data Source(s)

azurerm_eventhub_namespace

### Terraform Configuration Files

```hcl
resource "azurerm_eventhub_namespace" "ehns1" {
  name                = "ehissue-1"
  location            = data.azurerm_resource_group.rg.location
  resource_group_name = data.azurerm_resource_group.rg.name
  sku                 = "Standard"
  capacity            = 1
  zone_redundant      = true

  network_rulesets {
    default_action = "Deny"
    virtual_network_rule = [{
      subnet_id                                       = data.azurerm_subnet.vm-subnet.id
      # ignore_missing_virtual_network_service_endpoint = false
    }]
  }
}
```

### Debug Output/Panic Output

```shell
➜ terraform apply                                                                                                                                                                                                   
╷
│ Error: Incorrect attribute value type
│
│   on main.tf line 52, in resource "azurerm_eventhub_namespace" "ehns1":
│   52:     virtual_network_rule = [{
│   53:       subnet_id                                       = data.azurerm_subnet.vm-subnet.id
│   54:       # ignore_missing_virtual_network_service_endpoint = false
│   55:     }]
│     ├────────────────
│     │ data.azurerm_subnet.vm-subnet.id will be known only after apply
│
│ Inappropriate value for attribute "virtual_network_rule": element 0: attribute "ignore_missing_virtual_network_service_endpoint" is required.
╵
```

### Expected Behaviour

Either the documentaion or the behaviour should be corrected

### Actual Behaviour

As per the [docs ](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/eventhub_namespace#ignore_missing_virtual_network_service_endpoint), `ignore_missing_virtual_network_service_endpoint` is optional, but the error is thrown `"ignore_missing_virtual_network_service_endpoint" is required.`

**Comment by katbyte** [comment]
Thank you for taking the time to report this issue, and apologies that it has sat without a resolution for so long.

This bug was reported against an early major version of the AzureRM provider. The provider has since moved on to v5.x, and the affected functionality has often been significantly reworked across those major releases, so bug reports from legacy major versions prior to v4.0 are generally no longer actionable as written.

As part of a cleanup of historical issues we are closing bug reports that target legacy provider versions. If you are still running into this behaviour on the current major version, please open a new issue with up-to-date reproduction details (feel free to reference this one) so it can be triaged with current context.

Thank you again for the report, and for your patience.
